controllers/grid/users/author/AuthorGridHandler.inc.php

```diff
@@ -358,11 +407,15 @@ function updateAuthor($args, $request) {
 	 */
 	function deleteAuthor($args, $request) {
 		if (!$request->checkCSRF()) return new JSONMessage(false);
-
-		$submission = $this->getSubmission();
+		// Identify the submission Id
+		$submissionId = (int) $request->getUserVar('submissionId');
+		// Identify the author to be deleted
 		$authorId = (int) $request->getUserVar('authorId');
+		// Identify the version
+		$version = (int) $request->getUserVar('version');
+
 		$authorDao = DAORegistry::getDAO('AuthorDAO');
-		$authorDao->deleteById($authorId, $submission->getId());
+		$authorDao->deleteById($authorId, $submissionId, $version);
```

Comment:
I think it is good to use `$submission = $this->getSubmission();` because of the security -- at least the authorization policies control who has access to that submission. This way it would be maybe possible to delete any author of any submission, buy just providing their IDs as request parameters i.e. manipulating the form?